scality_manila_utils/export.py

```diff
@@ -0,0 +1,264 @@
+            log.debug("Export updated: %r", export)
+
+        self.exports[export_point] = export
+
+    def remove_client(self, export_point, host):
+        """
+        Remove access for a client to an export.
+
+        The filesystem is unexported if there are no clients left.
+
+        :param export_point: export to remove access from
+        :type export_point: string (unicode)
+        :param host: ip address, network or domain name for removal
+        :type host: string (unicode)
+        """
+        if export_point not in self.exports:
+            raise ExportException("No export point found for '{0:s}'".format(
+                                  export_point))
+
+        export = self.exports[export_point]
+        if host not in export.clients:
+            raise ExportException("'{0:s}' has no access defined for "
+                                  "'{1:s}'".format(export_point, host))
+
+        clients = export.clients
+        # If there are still clients after removal,
+        # replace the export with an updated version
+        if len(clients) > 1:
+            del clients[host]
+            self.exports[export_point] = Export(export_point, clients)
+        # Otherwise remove the export
+        else:
+            del self.exports[export_point]
+
+        log.debug("'%s' revoked from export '%s'", host, export_point)
+
+    @classmethod
+    def deserialize(cls, export_content):
+        """
+        Create an `ExportTable` from the contents of an /etc/exports file.
+
+        Each line of the exports file will be represented by a
+        :py:class:`scality_manila_utils.exports.Export`. Lines consisting of
+        whitespace only or that are comments will be ignored.
+
+        :param export_content: exports file contents split into a list of
+            strings
+        :type export_content: list of strings
+        :returns: a :py:class`scality_manila_utils.exports.ExportTable` object
+            with the exported filesystems
+        """
+        def strip_comment(line):
+            export, _, _ = line.partition('#')
+            return export
+
+        def is_blank(line):
+            stripped = line.strip()
+            return stripped == '' or stripped.startswith('#')
+
+        return ExportTable(
+            Export.deserialize(strip_comment(line))
+            for line in export_content
+            if not is_blank(line)
+        )
+
+    def serialize(self):
+        """
+        Serialize the `ExportTable` to a string following /etc/exports format.
+
+        :returns: string representation of the exports
+        """
+        return '\n'.join(
+            export.serialize() for export in self.exports.values()
+        )
+
+    def __eq__(self, other):
+        if not isinstance(other, ExportTable):
+            return NotImplemented
+
+        return self.exports == other.exports
+
+    def __ne__(self, other):
+        eq = self.__eq__(other)
+
+        if isinstance(eq, bool):
```

Comment:
Any reason why to prefer an instance check over `is` equality to `NotImplemented`?